Functions/xASL_quant_HadamardDecoding.m

```diff
@@ -0,0 +1,165 @@
+% EXAMPLE:      n/a
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+% Decoding Fields
+
+TimeEncodedMatrixType = xDecodingFields.TimeEncodedMatrixType;
+TimeEncodedMatrixSize = xDecodingFields.TimeEncodedMatrixSize;
+
+if isfield(xDecodingFields,'DecodingMatrix') && ~isempty(xDecodingFields.DecodingMatrix)
+    DecodingMatrix_input = xDecodingFields.DecodingMatrix;
+else
+    DecodingMatrix_input = [];
+end
+
+% #### For Walsh Decoding Matrix ####
+if strcmp(TimeEncodedMatrixType,'Walsh')
+    
+    if TimeEncodedMatrixSize == 4
+        
+        ASL_im = xASL_io_Nifti2Im(Encoded_ASL); % Load time-series nifti
+        
+        if ~isempty(DecodingMatrix_input) %if there's a Decoding Matrix from the data
+            DecodingMatrix = DecodingMatrix_input;
+        else
+            DecodingMatrix = [1 -1  1 -1;
+                              1 -1 -1  1;
+                              1  1 -1 -1];
+        end
+        
+    elseif TimeEncodedMatrixSize == 8
+        
+        ASL_im = xASL_io_Nifti2Im(Encoded_ASL); % Load time-series nifti
+        
+        if ~isempty(DecodingMatrix_input) %if there's a Decoding Matrix from the data
+            DecodingMatrix = DecodingMatrix_input;
+        else
+            DecodingMatrix = [1 -1  1 -1  1 -1  1 -1;
+                              1 -1  1 -1 -1  1 -1  1;
+                              1 -1 -1  1 -1  1  1 -1;
+                              1 -1 -1  1  1 -1 -1  1;
+                              1  1 -1 -1  1  1 -1 -1;
+                              1  1 -1 -1 -1 -1  1  1;
+                              1  1  1  1 -1 -1 -1 -1];
+        end
+    end
+    
+    % #### For Hadamard Decoding Matrix ####
+elseif strcmp(TimeEncodedMatrixType,'Hadamard')
+    
+    
+    
+    if TimeEncodedMatrixSize == 4
+        
+        ASL_im = xASL_io_Nifti2Im(Encoded_ASL); % Load time-series nifti
+        
+        if ~isempty(DecodingMatrix_input) %if there's a Decoding Matrix from the data
+            DecodingMatrix = DecodingMatrix_input;
+        else
+            DecodingMatrix = [1 -1  1 -1;
+                              1  1 -1 -1;
+                              1 -1 -1  1];
+        end
+        
+    elseif TimeEncodedMatrixSize == 8
+        
+        ASL_im = xASL_io_Nifti2Im(Encoded_ASL); % Load time-series nifti
+        
+        if ~isempty(DecodingMatrix_input) %if there's a Decoding Matrix from the data
+            DecodingMatrix = DecodingMatrix_input;
+        else
+            DecodingMatrix = [1 -1 -1  1 -1  1  1 -1;
+                              1  1 -1 -1 -1 -1  1  1;
+                              1 -1  1 -1 -1  1 -1  1;
+                              1  1  1  1 -1 -1 -1 -1;
+                              1 -1 -1  1  1 -1 -1  1;
+                              1  1 -1 -1  1  1 -1 -1;
+                              1 -1  1 -1  1 -1  1 -1];
+        end
+    end
+    
+end
+
+%% step-1: Reorder data
```

Comment:
Do you reorder the data (i.e., images/image volumes) here, or only the assumed order of the images?

Reply:
@BeatrizPadrela: just add another comment to make it more "clear" for the reviewer

Reply:
In dcm2nii, we order the data like this (PLDs first, TEs after), so also for Amnah's data it should be like this. So we reorder it for decoding